### PR TITLE
✨ [VIEW-1078] Allow user to click on the carousel item to navigate to other page using "router-link"

### PR DESCRIPTION
**Pull request by limhenry**
## Details
TL;DR: Allow user to click on the carousel item to navigate to other page using `router-link`.

The previous method was using Flickity's `staticClick`. However, this method won't work if the user want to open the link in new tab (by holding the Cmd/Ctrl button), or right click then select "Open Link in New Tab".

This can be done by using `router-link`. If there's no link, then the carousel item will fallback to `div`.

https://user-images.githubusercontent.com/4544770/183681412-7096a95d-e1b9-4295-a177-15fba46a8eb1.mov

## Related Ticket
[VIEW-1078](https://homeday.atlassian.net/browse/VIEW-1078)

## Related PR
https://github.com/homeday-de/customer-app/pull/2917

## Preview
- myHD: https://frontend-pulls.homeday.de/customer-app/2917
- Storybook: https://frontend-pulls.homeday.de/homeday-blocks/1011/

## Major Changes
- ✨`HdHallery`: Added `to` prop
- ✨`HdGalleryCarousel`, `HdGalleryMedia`: Added `to` prop and dynamic component (`div` and `router-link`)

### CODE REVIEW

**Review by aym3nb — APPROVED** [review]
Nicely done 👏 🚀
